src/search/FindUtils.js

```diff
@@ -352,11 +387,209 @@ define(function (require, exports, module) {
+            startingWorkingFileSet = [],
+            propertyName = "",
+            i = 0;
+        firstFile = firstFile || "";
+
+        // Create a working set path map which indicates if a file in working set is found in file list
+        for (i = 0; i < workingSetFiles.length; i++) {
+            workingSetFileFound[workingSetFiles[i].fullPath] = false;
+        }
+
+        // Remove all the working set files from the filtration list
+        fileSetWithoutWorkingSet = files.filter(function (key) {
+            if (workingSetFileFound[key] !== undefined) {
+                workingSetFileFound[key] = true;
+                return false;
+            }
+            return true;
+        });
+
+        //push in the first file
+        if (workingSetFileFound[firstFile] === true) {
+            startingWorkingFileSet.push(firstFile);
+            workingSetFileFound[firstFile] = false;
+        }
+        //push in the rest of working set files already present in file list
+        for (propertyName in workingSetFileFound) {
+            if (workingSetFileFound.hasOwnProperty(propertyName) && workingSetFileFound[propertyName]) {
+                startingWorkingFileSet.push(propertyName);
+            }
+        }
+        return startingWorkingFileSet.concat(fileSetWithoutWorkingSet);
+    }
+
+
+    /**
+     * Returns the path of the currently open file or null if there isn't one open
+     * @return {?string}
+     */
+    function getOpenFilePath() {
+        var currentDoc = DocumentManager.getCurrentDocument();
+        return currentDoc ? currentDoc.file.fullPath : null;
+    }
+
+    /**
+     * enable/disable instant search
+     * @param {boolean} disable true to disable node based search
+     */
+    function setInstantSearchDisabled(disable) {
+        instantSearchDisabled = disable;
+    }
+
+    /**
+     * if instant search is disabled, this will return true we can only do instant search through node
+     * @returns {boolean}
+     */
+    function isInstantSearchDisabled() {
+        return _prefNodeSearchDisabled() || _prefInstantSearchDisabled() || nodeSearchDisabled || instantSearchDisabled;
+    }
+
+    /**
+     * enable/disable node based search
+     * @param {boolean} disable true to disable node based search
+     */
+    function setNodeSearchDisabled(disable) {
+        if (disable) {
+            // only set disable. Enabling node earch doesnt mean we have to enable instant search.
+            setInstantSearchDisabled(disable);
+        }
+        nodeSearchDisabled = disable;
+    }
+
+    /**
+     * if node search is disabled, this will return true
+     * @returns {boolean}
+     */
+    function isNodeSearchDisabled() {
+        return _prefNodeSearchDisabled() || nodeSearchDisabled;
+    }
+
+    /**
+     * check if a search is progressing in node
+     * @returns {Boolean} true if search is processing in node
+     */
+    function isNodeSearchInProgress() {
+        if (nodeSearchCount === 0) {
```

Comment:
Linter says (confusing use of !)[http://derickbailey.com/2015/03/25/jshint-confusing-use-of-bang/].